Library/Homebrew/github_packages.rb

```diff
@@ -205,7 +203,7 @@ def upload_bottle(user, token, skopeo, formula_full_name, bottle_hash, dry_run:)
       formulae_dir = tag_hash["formulae_brew_sh_path"]
       documentation = "https://formulae.brew.sh/#{formulae_dir}/#{formula_name}" if formula_core_tap
 
-      tag = "#{version}.#{bottle_tag}#{rebuild}"
+      tag = "#{version_rebuild}.#{bottle_tag}"
```

Comment:
I would rather keep the `bottle_tag` before the `rebuild` so it is consistent with the bottle filenames.

Reply:
It’s typical from what I’ve seen with multi-architecture Docker images to use a Docker tag of `#{tag}-#{arch}` for the single architecture images, where `#{tag}` is the tag of the multi-architecture image. Putting the arch somewhere in the middle of the tag is a bit odd.

Even with the bottle filenames, I thought it was odd that the bottle tag was placed between the `revision` and the `rebuild`. With the Docker tags we have an opportunity to format them in a more consistent less idiosyncratic way.

Reply:
I prefer consistency with our existing bottles rather than consistency with Docker.

Reply:
The current bottle filename scheme is idiosyncratic, and I don't see a strong argument for making the Docker tag consistent with the bottle filename. I don't like the bottle filename scheme. I have wished many times that `*.bottle.tar.gz` would match all bottles. We have a chance to get it right now with Docker tags.

Reply:
> The current bottle filename scheme is idiosyncratic

I disagree.

> I don't see a strong argument for making the Docker tag consistent with the bottle filename.

I see consistency as its own argument.

> We have a chance to get it right now with Docker tags.

When are you going to use `*.bottle` or similar to match Docker tags? I don't see any logic to this.

Reply:
Package tags of
`hello_1-1`
`hello_1.big_sur.2`
`hello_1.x86_64_linux.2`
look more consistent and natural to me than
`hello_1-2`
`hello_1.big_sur.2`
`hello_1.x86_64_linux.2`

It's idiosyncratic that the architecture is in the middle of the version, between revision and rebuild.
It's inconsistent that the rebuild separator is `-` for the multi-architecture tag, and `.` for the single-architecture tag.

I hold this opinion strongly, but in the spirit of compromise on getting on with uploading bottles, I'm willing to defer if you feel equally strongly.

Reply:
*Edited* a copy-paste typo. Sorry if you read this in your e-mail and get confused.

Reply:
> It's idiosyncratic that the architecture is in the middle of the version, between revision and rebuild.

I disagree. The rebuild is a specific to the bottle tag rather than being anything inherent about the version. We've considered (and may consider again) having rebuilds independent per-bottle-tag which I think makes more sense with the existing versioning scheme.

> I hold this opinion strongly, but in the spirit of compromise and getting on with uploading bottles, I'm willing to defer if you feel equally strongly.

I do, thanks ❤️